sys/hacl/build.rs

```diff
@@ -25,34 +30,42 @@ fn includes(home_dir: &Path) -> Vec<String> {
 }
 
 fn append_simd128_flags(flags: &mut Vec<String>) {
-    // Platform detection
-    if cfg!(simd128) {
-        flags.push("-DSIMD128".to_string());
-        if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
-            flags.push("-msse4.1".to_string());
-            flags.push("-msse4.2".to_string());
-            flags.push("-mavx".to_string());
-        }
+    if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
+        flags.push("-msse4.1".to_string());
+        flags.push("-msse4.2".to_string());
+        flags.push("-mavx".to_string());
     }
 }
 
 fn append_simd256_flags(flags: &mut Vec<String>) {
-    // Platform detection
-    if cfg!(simd256) {
-        flags.push("-DSIMD256".to_string());
-        flags.push("-mavx2".to_string());
-    }
+    flags.push("-mavx2".to_string());
 }
 
 #[cfg(not(windows))]
-fn create_bindings(home_dir: &Path) {
-    let mut clang_args = includes(home_dir);
-    append_simd128_flags(&mut clang_args);
-    append_simd256_flags(&mut clang_args);
+fn create_bindings(platform: Platform, home_dir: &Path) {
+    let mut clang_args = includes(home_dir, "-I");
 
-    let bindings = bindgen::Builder::default()
+    let mut bindings = bindgen::Builder::default();
+    bindings = bindings
         // Header to wrap HACL headers
-        .header("c/include/hacl.h")
+        .header("c/include/hacl.h");
+
+    if platform.simd128 {
```

Comment:
There are some more platform dependent defines we should set.
The following values are usually set in `config.h`. We need to set them depending on the platform we are building for.
But let's file a follow up issue to tackle it in another PR and get a first version in that works.

- `TARGET_ARCHITECTURE`
- `HACL_CAN_COMPILE_UINT128`
- `HACL_CAN_COMPILE_VALE`
- `LINUX_NO_EXPLICIT_BZERO`

Reply:
I agree.

Reply:
Can you file an issue for it?

Reply:
Filed! https://github.com/cryspen/libcrux/issues/60